lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -0,0 +1,52 @@
+import importlib
+import os
+import sys
+
+from matplotlib.compat.subprocess import Popen
+import pytest
+
+
+# Minimal smoke-testing of the backends for which the dependencies are
+# PyPI-installable on Travis.  They are not available for all tested Python
+# versions so we don't fail on missing backends.
+#
+# Once the Travis build environment switches to Ubuntu 14.04, we should be able
+# to add wxagg (which has wheels for 14.04 but not for 12.04).
+
+
+def _get_available_backends():
+    backends = []
+    for module_name, backend in [
+            ("PyQt5", "qt5agg"),
+            ("Tkinter", "tkagg"),
+            ("tkinter", "tkagg")]:
+        try:
+            module = importlib.import_module(module_name)
+        except ImportError:
+            module = None
+        backends.append(
+            pytest.mark.skipif(
+                module is None,
+                reason="Could not import {!r}".format(module_name))(backend))
+    return backends
+
+
+_test_script = """\
+import sys
+from matplotlib import pyplot as plt
+
+fig = plt.figure()
+fig.canvas.mpl_connect("draw_event", lambda event: sys.exit())
+plt.show()
+"""
+
+
+@pytest.mark.skipif("DISPLAY" not in os.environ,
+                    reason="The DISPLAY environment variable is not set.")
+@pytest.mark.parametrize("backend", _get_available_backends())
+def test_backend(backend):
+    environ = os.environ.copy()
+    environ["MPLBACKEND"] = backend
+    proc = Popen([sys.executable, "-c", _test_script], env=environ)
+    # Empirically, 1s is not enough on Travis.
+    assert proc.wait(timeout=5) == 0
```

Comment:
bah, let's ignore py2.  that'll also let me handle @tacaswell's concern re: importing.